remove merging label when merge is cancelled

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98967

Adds a script to get rid of the "merging" label when a job is cancelled. 
At the moment this can create a race condition is someone cancels a job and starts a new one, though these cases should be pretty rare especially in cases where its from a new merge command.

PaliC committed:
remove merging label when merge is cancelled

[ghstack-poisoned]

PaliC committed:
Update on "remove merging label when merge is cancelled"




[ghstack-poisoned]

malfet commented on .github/scripts/on_cancel_merge.py:
```diff
@@ -0,0 +1,24 @@
+    org, project = repo.gh_owner_and_name()
+    pr_num = args.pr_num
+
+    GitHubPR(org, project, pr_num).remove_label(MERGE_IN_PROGRESS_LABEL)
```
Please add a new line
```suggestion
    GitHubPR(org, project, pr_num).remove_label(MERGE_IN_PROGRESS_LABEL)

```

malfet commented on .github/workflows/trymerge.yml:
```diff
@@ -75,6 +75,7 @@ jobs:
         run: |
           set -ex
+          python3 .github/scripts/on_cancel_merge.py "${PR_NUM}"
           python3 .github/scripts/comment_on_pr.py "${PR_NUM}" "merge"
```
Can we merge those scripts together? (If `comment_on_pr.py` already imports GitHub class

PaliC replied:
I was originally going to do that, however, comment on pr is used in other bits which are not related to merging.

PaliC committed:
Update on "remove merging label when merge is cancelled"


Adds a script to get rid of the "merging" label when a job is cancelled. 
At the moment this can create a race condition is someone cancels a job and starts a new one, though these cases should be pretty rare especially in cases where its from a new merge command.

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge

PaliC commented:
@pytorchbot merge -rebase

PaliC commented:
@pytorchbot merge -r
